packages/spanner/src/database.js

```diff
@@ -713,45 +713,43 @@ Database.prototype.runStream = function(query, options) {
  * });
  *
  * //-
- * // If the callback is omitted, we'll return a Promise.
+ * // To use promises, simply return a Promise.
  * //-
- * database.runTransaction()
- *   .then(function(data) {
- *     var transaction = data[0];
- *
- *     // Run a transactional query.
- *     return transaction.run('SELECT * FROM Singers')
- *       .then(function() {
- *         // Queue a mutation (note there is no callback passed to `insert`).
- *         transaction.insert('Singers', {
- *           SingerId: 'Id3b',
- *           Name: 'Joe West'
- *         });
- *
- *         // Commit the transaction.
- *         return transaction.commit();
- *       });
- *   })
- *   .then(function() {
- *     // Transaction committed successfully.
+ * database.runTransaction(function(err, transaction) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // Run a transactional query.
+ *   return transaction.run('SELECT * FROM Singers').then(function() {
+ *     // Queue a mutation (note there is no callback passed to `insert`).
+ *     transaction.insert('Singers', {
+ *       SingerId: 'Id3b',
+ *       Name: 'Joe West'
+ *     });
+ *     // Commit the transaction.
+ *     return transaction.commit();
```

Comment:
I just realized I did not reply to something that @vkedia asked:

> @lukesneeringer I don't understand your comment. Can you please explain with an example? 

An example of multiple statements in a transaction is:

```es6
transaction.runTransaction()
  .then(txn => txn.insert('...data to insert...'))
  .then(txn => txn.update('...data to change...'))
  .then(txn => txn.insert('...other data to insert...'))
  .then(txn => txn.commit());
```

> The Promise that the function returns should resolve only after they have done all they want with the transaction.

I think you may misunderstand what promises are for. They are not a tool for "wrapping" arbitrary amounts of behavior like a context manager; they are a way of idiomatically waiting on _a single_ asychronous event. Chaining a bunch of `.then` calls is common (see my example).

So, there is absolutely no guarantee that the resolution of a specific promise means that you are done with the transaction. In fact, the opposite is true -- it would be entirely idiomatic for the promise to resolve when the transaction is _successfully opened_, before you execute a single statement (whether the code actually does that I am unsure).